kimchi/src/circuits/polynomials/foreign_field_add/witness.rs

```diff
@@ -183,11 +186,45 @@ pub fn create<F: PrimeField>(
     for w in &mut witness {
         w.extend(std::iter::repeat(F::zero()).take(2));
     }
-    init_foreign_field_fin_rows(&mut witness, offset, num, [bound_carry_lo, bound_carry_mi]);
+    init_foreign_field_fin_rows_from_multi_range_check(
+        &mut witness,
+        offset,
+        num,
+        [bound_carry_lo, bound_carry_mi],
+    );
 
     witness
 }
 
+/// Create witness for bound computation addition gate
+pub fn extend_witness_bound_addition<F: PrimeField>(
+    witness: &mut [Vec<F>; COLUMNS],
+    limbs: &[F; 3],
```

Comment:
I found it cleaner without it because I was able to switch between representations less often and  more easily this way. 
 Notice there's less duplicate copies of variables with some in field and some in bigunint and some in foreign element.  Instead I just use these simple helpers where needed.